cli/client/client.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// Client .
+type Client interface {
+	GET(apiURL string, debug bool, extraHeaders http.Header, result interface{}) error
+	POST(apiURL string, body io.Reader, debug bool, headers http.Header, result interface{}) error
+	PUT(apiURL string, body io.Reader, debug bool, headers http.Header, result interface{}) error
+	DELETE(apiURL string, debug bool, extraHeaders http.Header, result interface{}) error
+	SetHTTPClient(client *http.Client)
+	SetRegistryToken(token string)
+	SetRegistryURL(baseURL string)
+	NewHeader() http.Header
+}
+
+// A8client .
+type A8client struct {
+	url        string
+	token      string
+	httpClient *http.Client
+}
+
+// NewClient .
+func NewClient(url, token string, HTTPclient *http.Client) Client {
+	// If not http client has been defined, use the default HTTP client
+	if HTTPclient == nil {
+		HTTPclient = &http.Client{
+			Timeout: 30 * time.Second,
+		}
+	}
+
+	return &A8client{
+		url:        url,
+		token:      token,
+		httpClient: HTTPclient,
+	}
+}
+
+// SetRegistryURL .
+func (c *A8client) SetRegistryURL(baseURL string) {
+	c.url = baseURL
+}
+
+// SetRegistryToken .
+func (c *A8client) SetRegistryToken(token string) {
+	c.token = token
+}
+
+// SetHTTPClient .
+func (c *A8client) SetHTTPClient(client *http.Client) {
```

Comment:
I think the comment here has nothing to do with code duplication. @elevran is talking about handling error conditions (whether or not there is code duplication).